src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2475,90 +2475,49 @@ internal static int IndexOfAnyChar(ref char searchSpace, char value0, char value
             }
             else if (Vector512.IsHardwareAccelerated && length >= Vector512<TValue>.Count)
             {
-                Vector512<TValue> current, values = Vector512.Create(value);
-                nint offset = length - Vector512<TValue>.Count;
-
-                // Loop until either we've finished all elements -or- there's one or less than a vector's-worth remaining.
-                while (offset > 0)
-                {
-                    current = Vector512.LoadUnsafe(ref searchSpace, (nuint)(offset));
-
-                    if (TNegator.HasMatch(values, current))
-                    {
-                        return ComputeLastIndex(offset, TNegator.GetMatchMask(values, current));
-                    }
-
-                    offset -= Vector512<TValue>.Count;
-                }
-
-                // Process the first vector in the search space.
-
-                current = Vector512.LoadUnsafe(ref searchSpace);
-
-                if (TNegator.HasMatch(values, current))
-                {
-                    return ComputeLastIndex(offset: 0, TNegator.GetMatchMask(values, current));
-                }
+                return SimdImpl<Vector512<TValue>>(ref searchSpace, value, length);
```

Comment:
I love how we can now share the same implementation across the three uses.

I do wonder whether it'd be worth implementing a few more such reductions in this PR, just to better help validate before unleashing it on others and the rest of corelib. It's possible all bases are covered, but we'll often find interesting things in the 2nd and 3rd uses of an abstraction that we didn't encounter in the 1st.

Reply:
I think it'd be better to get this in, so we can get the early perf numbers (which will take a week).

Especially since this is staying internal only, we can make adjustments if any users find them as we update other APIs to utilize it.

Ideally we do this with a couple people before we try and touch "everything". The ones I expect we're "missing" are net new APIs covering core concepts over `ExtractMostSignificantBits`, such as `IndexOfFirst/LastMatch`